src/retrieve-data-helpers/smart-launch.js

```diff
@@ -0,0 +1,48 @@
+import { signalSuccessFhirServerRetrieval } from '../actions/fhir-server-actions';
+import { signalSuccessSmartAuth, signalFailureSmartAuth } from '../actions/smart-auth-actions';
+import store from '../store/store';
+
+/**
+ * Handle SMART launch (Oauth2 response) accordingly if Sandbox is launched as a SMART on FHIR app.
+ * Store relevant data passed in.
+ * @returns {Promise} - Promise to resolve elsewhere
+ */
+function smartLaunchPromise() {
+  return new Promise((resolve, reject) => {
+    const onAuthFailure = () => {
+      console.log('Launched openly and unsecured');
+      store.dispatch(signalFailureSmartAuth());
+      return reject();
+    };
+
+    const onAuthSuccess = (smart) => {
+      console.log('Launched as a SMART app');
+      if (smart.tokenResponse) {
+        let fhirBaseUrl;
+        if (smart.server && smart.server.serviceUrl) {
+          fhirBaseUrl = smart.server.serviceUrl;
+          const conformanceQuery = smart.api.conformance({});
+          const deferredToPromise = Promise.resolve(conformanceQuery);
+          deferredToPromise.then((result) => {
+            store.dispatch(signalSuccessSmartAuth(smart));
+            store.dispatch(signalSuccessFhirServerRetrieval(fhirBaseUrl, result.data));
+            return resolve(result.data);
+          }).catch((err) => {
+            console.error('Failed to get metadata from secured FHIR server. Launching sandbox openly', err);
```

Comment:
You could clean up the console.error's by passing in the error message to onAuthFailure and report the error in that method. 
Not a necessary change but this should reduce your effort in jest function mocks that you would need to do for console.error function's.